internal/dag/builder.go

```diff
@@ -487,6 +487,17 @@ func (b *Builder) computeHTTPProxy(proxy *projcontour.HTTPProxy) {
 			svhost := b.lookupSecureVirtualHost(host)
 			svhost.Secret = sec
 			svhost.MinProtoVersion = MinProtoVersion(proxy.Spec.VirtualHost.TLS.MinimumProtocolVersion)
+
+			// Fill in DownstreamValidation when external client validation is enabled
+			if tls.ClientValidation != nil {
+				dv, err := b.lookupDownstreamValidation(tls.ClientValidation,
+					proxy.Namespace,
+					fmt.Sprintf("HTTPProxy: %q", proxy.Name))
```

Comment:
I made some minor cleanups to `lookupUpstreamValidation`, so here you should follow that pattern and add status message context in the caller.

Reply:
Fixed.